thunder/executors/nvfuserex_impl.py

```diff
@@ -2194,3 +2194,28 @@ def map_inside_replacement(x: Any) -> None:
     elapsed_time_millis = elapsed_time_ns // 1000000
     rrctrace.set_provenance(TraceProvenance(f"Remove redundant casts (took {elapsed_time_millis} milliseconds)"))
     return rrctrace
+
+
+def _matmul_check(
+    a: TensorProxy,
+    b: TensorProxy,
+) -> bool:
+    enable_matmul: None | bool = get_compile_option("nv_enable_matmul", "Enable nvFuser matmul.")
```

Comment:
This is where you should check nvfuser version and reject matmul for older nvfuser.